core/src/main/java/org/mule/config/ChainedThreadingProfile.java

```diff
@@ -83,7 +83,7 @@ public ChainedThreadingProfile(ThreadingProfile delegate)
      */
     public ChainedThreadingProfile(ThreadingProfile delegate, boolean dynamic)
     {
-        if (!dynamic)
+        if (!(dynamic || delegate instanceof DirectThreadingProfile))
```

Comment:
I'm not sure there is a direct relationship between the two.  The issue here is that currently threadProfile is always copied, get the copy mechanism (copy to immutableThreadingProfile) over-rides all implementation behaviour because it does not delegate.  So what i have had to do (to avoid bigger changes) is simply not copy in the case of the direct implementation but continue copying as today in the other cases.